File: Ordenacion/Directos/Bubblesort/bubblesort.py
def bubblesort(vector):
    n = len(vector)  # n es el tamaño del vector
    swapped = True
    for i in range(n - 1):
        if not swapped:  # Si no hubo intercambios, el vector ya está ordenado
            break
        swapped = False
        for j in range(n - i - 1):
            if vector[j] > vector[j + 1]:
                # Intercambiar los elementos
                vector[j], vector[j + 1] = vector[j + 1], vector[j]
                swapped = True

File: Ordenacion/Directos/Bubblesort/bubblesort.cpp
using namespace std;

void bubblesort(int vector[], int n) { //n es el tamaño del vector
    int aux;
    bool swapped = true;
    for (int i = 0; i < n - 1 && swapped; i++) {
        swapped = false;
        for (int j = 0; j < n - i - 1; j++) {
            if (vector[j] > vector[j + 1]) {
                aux = vector[j];
                vector[j] = vector[j + 1];
                vector[j + 1] = aux;
                swapped = true;
            }
        }
    }
}